jaeger-core/src/main/java/com/uber/jaeger/senders/HttpSender.java

```diff
@@ -44,41 +43,56 @@
   private static final TProtocolFactory PROTOCOL_FACTORY = new Factory();
   private static final TSerializer SERIALIZER = new TSerializer(PROTOCOL_FACTORY);
 
-  private final HttpClient httpClient = new DefaultHttpClient();
-  private final URI collectorUri;
+  private final OkHttpClient httpClient = new OkHttpClient();
+  private final HttpUrl collectorUrl;
+  private static final int ONE_MB_IN_BYTES = 1048576;
 
   /**
-   * @param endpoint Jaeger REST endpoint consuming jaeger.thrift, e.g http://localhost:14268/api/traces
-   * @param maxPacketSize max packet size
+   * @param endpoint Jaeger REST endpoint consuming jaeger.thrift, e.g
+   * http://localhost:14268/api/traces
    */
-  public HttpSender(String endpoint, int maxPacketSize) {
-    super(PROTOCOL_FACTORY, maxPacketSize);
-    this.collectorUri = constructCollectorUri(endpoint);
+  public HttpSender(String endpoint) {
+    this(endpoint, ONE_MB_IN_BYTES);
+  }
+
+  /**
+   * @param endpoint Jaeger REST endpoint consuming jaeger.thrift, e.g
+   * http://localhost:14268/api/traces
+   * @param maxPayloadBytes max bytes to serialize as payload
+   */
+  public HttpSender(String endpoint, int maxPayloadBytes) {
+    this(HttpUrl.parse(String.format("%s?%s", endpoint, HTTP_COLLECTOR_JAEGER_THRIFT_FORMAT_PARAM)), maxPayloadBytes);
+  }
+
+  /**
+   * @param url Jaeger url accepting jaeger.thrift. This needs to include query params, etc.
+   * @param maxPayloadBytes max bytes to serialize as payload
+   */
+  public HttpSender(HttpUrl url, int maxPayloadBytes) {
+    super(PROTOCOL_FACTORY, maxPayloadBytes);
+    if (url == null) {
+      throw new IllegalArgumentException("Could not parse url.");
+    }
+    this.collectorUrl = url;
   }
 
   @Override
-  public void send(Process process, List<com.uber.jaeger.thriftjava.Span> spans) throws TException {
+  public void send(Process process, List<Span> spans) throws TException {
     Batch batch = new Batch(process, spans);
     byte[] bytes = SERIALIZER.serialize(batch);
 
+    RequestBody body = RequestBody.create(MediaType.parse("application/x-thrift"), bytes);
+    Request request = new Request.Builder().url(collectorUrl).post(body).build();
```

Comment:
as I recall, the maxPacketSize is only relevant to the base class that decides when to send the batch vs. just append the span to the buffer. The actual request could be >1Mb.

Btw, where did the 1Mb number come from? Previous version of this class didn't seem to have such default, so why do we need to introduce it? There's no particular meaning in 1MB, it's really up to the user of this class to decide how much or little they should be batching.